### PR TITLE
Feature/116 ansible refactoring part1

### PR DESCRIPTION
**Pull request by ltoning**
- refactor code
- added roles : hosts, createusers, dnsmasq
- variablized role configs into vars file
- removed udhcpd and use dhcpd
-  fixed bug #110
- variablized hostapd config into vars file
- corrected readme as it's incorrect referenced the elimupi-connect
- update readme using latest buster Raspios Buster based release 2023-05-03
- updated version to 3.4.0

Fully tested working on atest buster Raspios Buster based release 2023-05-03

### CODE REVIEW

File: ansible/roles/common/tasks/setup-wifi.yml [hunk 1]
```diff
@@ -33,14 +28,31 @@
...
-    mode: '0755' 
+    mode: '0744'
+  with_items:
+    - v4
+
+- name: "Copy hs-iptables.service file"
+  ansible.builtin.template:
+    src: "hs-iptables.service.j2"
+    dest: "/etc/systemd/system/hs-iptables.service"
+    mode: "0755"
```
Comment (alonsovidales) [inline ansible/roles/common/tasks/setup-wifi.yml:45]: nit: We could go by 744 here as in 37

**Review by alonsovidales — APPROVED** [review]
looks good 👍

File: README.md [hunk 2]
```diff
@@ -66,33 +67,38 @@ The installation will take approximately 20 minutes to finish and there shouldn'
...
 
  - Create a image copy of the SD-card with the ElimuPi software installed using instructions at https://beebom.com/how-clone-raspberry-pi-sd-card-windows-linux-macos/
- -  Shrink cloned the image file ( Linux Only) using [PiShrink](https://github.com/Drewsif/PiShrink).
- - gzip the shrinked image file.
- - Name the image: **ElimuPi_Connect_Image_2022-09-06-raspios-bullseye-arm64_lite_Release_<-Version->.img.gz** e.g
+ -  Shrink cloned the image file ( Linux Only) using [PiShrink](https://github.com/Drewsif/PiShrink) and the compression option -Za 
+ - Name the image: **ElimuPi_Image_2023-05-03-raspios-bullseye-arm64_lite_Release_<-Version->.img.xz** e.g
 
-This should produce a 2.8GB gzipped compressed Elimupi image file.
+This should produce a xz compressed Elimupi image file.
```
Comment (marcobruining) [inline README.md:73]: Did you intend to have xz here, or should there be a value indicating size?

File: ansible/roles/common/tasks/setup-wifi.yml [hunk 3]
```diff
@@ -49,5 +61,5 @@
     daemon_reload: yes
     masked: no
   with_items:
-    - udhcpd
+    - dhcpcd
```
Comment (marcobruining) [inline ansible/roles/common/tasks/setup-wifi.yml:64]: Should this be dhcpcd5? Like in the vars.yaml is recorded as a variable?

Comment (ltoning) [reply ansible/roles/common/tasks/setup-wifi.yml:64]: no is correct.
dhcpcd5 is the package name
dhcpcd is the service name

File: ansible/roles/common/tasks/setup-wifi.yml (hunk 1 above)
Comment (ltoning) [reply ansible/roles/common/tasks/setup-wifi.yml]: Agree. changed it.

**Review by marcobruining — APPROVED** [review]

File: ansible/roles.yml [hunk 4]
```diff
@@ -1,2 +1,5 @@
+- src: singleplatform-eng.users
```
Comment (alonsovidales) [inline ansible/roles.yml:1]: Some of these roles are missing:
```
$ansible-playbook -i ./inventory.yml playbook-raspberrypi.yml --ask-vault-pass -vvv
ansible-playbook [core 2.14.3]
  config file = /Users/avidales/personal/elimupi/ansible/ansible.cfg
  configured module search path = ['/Users/avidales/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/homebrew/Cellar/ansible/7.3.0/libexec/lib/python3.11/site-packages/ansible
  ansible collection location = /Users/avidales/.ansible/collections:/usr/share/ansible/collections
  executable location = /opt/homebrew/bin/ansible-playbook
  python version = 3.11.2 (main, Feb 16 2023, 02:52:13) [Clang 13.0.0 (clang-1300.0.29.30)] (/opt/homebrew/Cellar/ansible/7.3.0/libexec/bin/python3.11)
  jinja version = 3.1.2
  libyaml = True
Using /Users/avidales/personal/elimupi/ansible/ansible.cfg as config file
Vault password:
host_list declined parsing /Users/avidales/personal/elimupi/ansible/inventory.yml as it did not pass its verify_file() method
script declined parsing /Users/avidales/personal/elimupi/ansible/inventory.yml as it did not pass its verify_file() method
Parsed /Users/avidales/personal/elimupi/ansible/inventory.yml inventory source with yaml plugin
ERROR! the role 'singleplatform-eng.users' was not found in /Users/avidales/personal/elimupi/ansible/roles:/Users/avidales/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/Users/avidales/personal/elimupi/ansible

The error appears to be in '/Users/avidales/personal/elimupi/ansible/playbook-raspberrypi.yml': line 7, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  roles:
    - singleplatform-eng.users
      ^ here
```

**Comment by ltoning** [comment]
Did you run ansible-galaxy install -r roles.yml ?

Op za 11 nov. 2023 20:24 schreef Alonso Vidales ***@***.***>:

**Comment by alonsovidales** [comment]
Ah, nope, now it works, thanks!

**Pull request merged**